### PR TITLE
Remove some `dynamic's.

### PR DESCRIPTION
**Pull request by Hixie**
...and add in some `!`s that were implied previously. Dynamic really is just a way to hide danger...

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Hixie** [comment]
cc @goderbauer 

I'm not attached to this PR, I was doing something unrelated and ended up with it for... reasons.

**Review by goderbauer — APPROVED** [review]
LGTM!

I am all for this! Looks like it is failing a check, though.

**Comment by Hixie** [comment]
Yeah I'll revert some of these to `dynamic` since they're public APIs.

**Comment by Hixie** [comment]
Updated, including some documentation changes. Will land if it goes green before you add another comment. :-)

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 1]
```diff
@@ -188,9 +194,20 @@ class JSONMethodCodec implements MethodCodec {
 ///  * [List]s of supported values
 ///  * [Map]s from supported values to supported values
 ///
-/// Decoded values will use `List<dynamic>` and `Map<dynamic, dynamic>`
+/// Decoded values will use `List<Object?>` and `Map<Object?, Object?>`
 /// irrespective of content.
 ///
+/// The type returned from `decodeMessage` is `dynamic` (not `Object?`), which
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/message_codecs.dart:200]: Same comments as above.

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 2]
```diff
@@ -71,7 +71,13 @@ class StringCodec implements MessageCodec<String?> {
 /// null/nil for null, and identical to what would result from decoding a
 /// singleton JSON array with a Boolean, number, or string value, and then
 /// extracting its single element.
-class JSONMessageCodec implements MessageCodec<dynamic> {
+///
+/// The type returned from `decodeMessage` is `dynamic` (not `Object?`), which
+/// means *no type checking is performed on its return value*. In is strongly
+/// recommended that the return value be immediately cast to a known type to
+/// prevent runtime errors due to typos that the type checked could otherwise
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/services/message_codecs.dart:78]: yeah that was a typo for "type checker"

**Comment by Hixie** [comment]
@goderbauer PTAL, the tests are green but I had to make some changes to how the generics are done that might be... exciting to consider.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**